### PR TITLE
Branch deep link integration

### PR DESCRIPTION
**Pull request by vinothvino42**
Ticket: https://github.com/EnsembleUI/ensemble/issues/1046

**Sample EDL**

```yaml

View:
  styles:
    scrollableView: true
  header:
    title: Deferred Deep Link SDK

  body:
    Column:
      styles: { gap: 16, padding: 24 }
      children:
        - Button:
            label: Init Branch SDK
            onTap:
              deeplinkInit:
                provider: branch
                options:
                  useTestKey: true
                  enableLog: true
                  disableTrack: false
                onSuccess: |
                  //@code
                  console.log("BranchSDK:: initialized successfully");
                onLinkReceived: |
                  //@code
                  console.log("BranchSDK:: Branch Link Received: " + event.data.link);
                onError: |
                  //@code
                  console.log("BranchSDK:: Failed to initialize" + event.error);

        - Text:
            id: branchLinkId
            text: Branch Link

        - Button:
            label: Create Branch Link
            onTap:
              createDeeplink:
                provider: branch
                universalProps:
                  id: item/12345
                  title: My Content Title
                  contentDescription: My Content Description
                  imageUrl: https://ensembleui.com/mycontent-123.png
                  contentSchema: product
                linkProps:
                  channel: ensembleui
                  feature: Sharing
                  campaign: content launch
                  stage: New User
                  tags: [data, fsdf, sdd]
                  controlParams:
                    ref: 1234
                    name: Ensemble
                onSuccess: |
                  //@code
                  console.log("BranchSDK:: Link created successfully: " + event.data.result);
                  branchLinkId.text = "Branch Link - " + event.data.result;
                onError: |
                  //@code
                  console.log("BranchSDK:: Failed to create a link" + event.error);

```

### CODE REVIEW

File: lib/framework/action.dart [hunk 1]
```diff
@@ -937,6 +939,10 @@ enum ActionType {
   callExternalMethod,
   invokeHaptic,
   callNativeMethod,
+  branchLinkInit,
+  branchLinkValidate,
+  branchLinkCreateDeeplink,
+  branchLinkCreateDeeplinkWithShareSheet,
```
Comment (snehmehta) [inline lib/framework/action.dart:945]: do we need `branchLinkCreateDeeplinkWithShareSheet`

Comment (vinothvino42) [reply lib/framework/action.dart:945]: Removed it

File: lib/framework/action.dart [hunk 2]
```diff
@@ -937,6 +939,10 @@ enum ActionType {
   callExternalMethod,
   invokeHaptic,
   callNativeMethod,
+  branchLinkInit,
```
[the remark this answers is not in the document]
Comment (vinothvino42) [reply lib/framework/action.dart:942]: Okay sneh, I'll update and let you know

**Comment by vinothvino42** [comment]
@snehmehta I updated it. Can you review it again?

**Review by snehmehta — APPROVED** [review]

**Pull request merged**